### PR TITLE
More auto-fixing of corelib (primarily mono/netcore)

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/System/WeakReference.T.Mono.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ private void Create(T target, bool trackResurrection)
         {
             if (trackResurrection)
             {
-                trackResurrection = true;
+                this.trackResurrection = true;
```
Comment (stephentoub) [inline src/mono/netcore/System.Private.CoreLib/src/System/WeakReference.T.Mono.cs]: This looked like a real bug.  Please let me know if something different should be done.

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs [hunk 2]
```diff
@@ -524,14 +522,14 @@ public int GetUpperBound(int dimension)
         [Intrinsic]
         private void GetGenericValueImpl<T>(int pos, out T value)
         {
-            var self = this;
+            Array? self = this;
```
Comment (marek-safar) [inline src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs]: Why Array?

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs [hunk 3]
```diff
@@ -524,14 +522,14 @@ public int GetUpperBound(int dimension)
...
             GetGenericValue_icall(ref self, pos, out value);
         }
 
         [Intrinsic]
         private void SetGenericValueImpl<T>(int pos, ref T value)
         {
-            var self = this;
+            Array? self = this;
```
Comment (marek-safar) [inline src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs]: Same as above

File: src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/StackTrace.Mono.cs [hunk 4]
```diff
@@ -60,7 +60,7 @@ private void InitializeForCurrentThread(int skipFrames, bool needFileInfo)
 
         private void InitializeForException(Exception e, int skipFrames, bool needFileInfo)
         {
-            var frames = get_trace(e, skipFrames, needFileInfo);
+            MonoStackFrame[]? frames = get_trace(e, skipFrames, needFileInfo);
```
Comment (marek-safar) [inline src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/StackTrace.Mono.cs]: Looks like wrong `?`

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs (hunk 2 above)
Comment (stephentoub) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs]: Looks like an unnecessary annotation added by the fixer.  Opened https://github.com/dotnet/roslyn/issues/42763.

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs (hunk 3 above)
Comment (stephentoub) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs]: Ditto

File: src/mono/netcore/System.Private.CoreLib/src/System/WeakReference.T.Mono.cs (hunk 1 above)
Comment (akoeplinger) [reply src/mono/netcore/System.Private.CoreLib/src/System/WeakReference.T.Mono.cs]: Yes looks like a bug that goes back to https://github.com/mono/mono/commit/30a2eeb98275a1432dc990ef6beee354e8bb2842.

I checked that we don't have the issue in the classic mono/mono implementation.

File: src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/StackTrace.Mono.cs (hunk 4 above)
Comment (stephentoub) [reply src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/StackTrace.Mono.cs]: Yup, thanks.

**Pull request merged**